### PR TITLE
[FEATURE] Add support for standard rst and sphinx text roles

### LINKED ISSUE #78: missing support for many inline markup items - like :kbd:
We know about `:ref:`and `:doc:`, but there are actually a bunch of other `:something:` inline syntaxes supported by Sphinx - http://www.sphinx-doc.org/es/stable/markup/inline.html#cross-referencing-documents

We use at least one in the symfony docs - `:kbd:`. I don't think that we should go and implement *every* one of these (there are a lot) - but it looks like we *do* need some systematic way of handling this `:something:` syntax. If I'm not mistaken, it looks like we currently treat *anything* matching `:*:` as a reference: https://github.com/doctrine/rst-parser/blob/8e8aa20eb072e69f557e4a49c4db0b50928a4ef6/lib/Span/SpanProcessor.php#L107

### PR DESCRIPTION
**Pull request by linawolf**
Make standard wrapper text roles that only need to be instantiated, add example HTML markup. Allow textrole registration by Format

resolves #78

See https://docs.typo3.org/m/typo3/docs-how-to-document/main/en-us/WritingReST/Reference/Code/InlineCode.html

https://docutils.sourceforge.io/docs/ref/rst/roles.html#toc-entry-3

### CODE REVIEW

**Review by greg0ire — APPROVED** [review]

File: lib/HTML/HTMLFormat.php [hunk 1]
```diff
@@ -44,6 +47,28 @@ public function getDirectives(): array
...
+            new WrapperTextRole('kbd', '<kbd class="kbd docutils literal notranslate">%s</kbd>'),
+            new WrapperTextRole('mailheader', '<em class="mailheader">%s</em>'),
+            new WrapperTextRole('math', '<math>%s</math>'),
+            new WrapperTextRole('emphasis', '<em>%s</em>'),
+            new WrapperTextRole('literal', '<span class="pre">%s</span>'),
+            new WrapperTextRole('strong', '<em class="strong">%s</em>'),
+            new WrapperTextRole('subscript', '<sub>%s</sub>', ['sub']),
+            new WrapperTextRole('superscript', '<sup>%s</sup>', ['sup']),
+            new WrapperTextRole('title-reference', '<cite>%s</cite>', ['t', 'title']),
+            new DefinitionTextRole('abbreviation', '<abbr title="%2$s">%1$s</abbr>', ['abbr']),
```
Comment (SenseException) [inline lib/HTML/HTMLFormat.php]: Where does the HTML representation of the roles come from? Did I miss something in the links?

**Review by SenseException — COMMENTED** [review]
This PR looks fine and ready to merge, but there's one point about the HTML I don't understand. Wrote it in a review-comment.

File: lib/HTML/HTMLFormat.php (hunk 1 above)
Comment (linawolf) [reply lib/HTML/HTMLFormat.php]: RST does not specify any HTML representations as RST can be transformed into many different Formats. So I took those we are doing in TYPO3 or you have a better idea?

Comment (SenseException) [reply lib/HTML/HTMLFormat.php]: Most of the tags make sense as they are the equivalent of the role. I'm confused about `<em class="strong">%s</em>` for strong. I'd expect the `<strong>` tag.

File: lib/HTML/HTMLFormat.php [hunk 2]
```diff
@@ -44,6 +47,28 @@ public function getDirectives(): array
...
+            new WrapperTextRole('command', '<strong class="command">%s</strong>'),
+            new WrapperTextRole('dfn', '<em class="dfn">%s</em>'),
+            new WrapperTextRole('file', '<span class="pre file">%s</span>'),
+            new WrapperTextRole('guilabel', '<span class="guilabel">%s</span>'),
+            new WrapperTextRole('kbd', '<kbd class="kbd docutils literal notranslate">%s</kbd>'),
+            new WrapperTextRole('mailheader', '<em class="mailheader">%s</em>'),
+            new WrapperTextRole('math', '<math>%s</math>'),
+            new WrapperTextRole('emphasis', '<em>%s</em>'),
+            new WrapperTextRole('literal', '<span class="pre">%s</span>'),
+            new WrapperTextRole('strong', '<em class="strong">%s</em>'),
```
Comment (linawolf) [inline lib/HTML/HTMLFormat.php]:
```suggestion
            new WrapperTextRole('strong', '<strong>%s</strong>'),
```

File: lib/HTML/HTMLFormat.php (hunk 1 above)
Comment (linawolf) [reply lib/HTML/HTMLFormat.php]: thats true, changed it to stron

**Review by greg0ire — APPROVED** [review]

**Review by SenseException — APPROVED** [review]

**Comment by SenseException** [comment]
@linawolf Thank you

**Pull request merged**